src/main/java/software/amazon/cloudformation/proxy/AmazonWebServicesClientProxy.java

```diff
@@ -443,9 +443,12 @@ public Logger getLogger() {
         request.setRequestCredentialsProvider(v1CredentialsProvider);
 
         try {
-            ResultT respose = requestFunction.apply(request);
-            logRequestMetadata(request, respose);
-            return respose;
+            ResultT response = requestFunction.apply(request);
+            if(response != null) {
+                logRequestMetadata(request, response);
+            }
+
+            return response;
```

Comment:
will that work? dont you get an exception if you trying to assign to a value void return? this wont give you exception on the compilation time but might error out on runtime